ansible/roles/atmo-ssh-setup/tasks/ssh-setup.yml

```diff
@@ -0,0 +1,42 @@
+---
+
+- name: Directly check linux distribution
+  stat: path=/etc/redhat-release
+  register: linux_distro
+
+- name: Gather OS-specific variables
+  include_vars: "{{ item.var }}"
+  when: "{{ item.condition }}"
+  with_items:
+    - { var: 'Ubuntu.yml', condition: "use_remote_user == 'ubuntu' or not linux_distro.stat.exists" }
+    - { var: 'CentOS.yml', condition: "use_remote_user == 'centos' or linux_distro.stat.exists" }
+  tags: vars
+
+- name: Make sure ssh directory exists with correct permissions
+  file:
+    path: /root/.ssh/
+    state: directory
+    mode: "0700"
+
+- name: Make changes to /etc/ssh/sshd_config
+  lineinfile:
+    dest: /etc/ssh/sshd_config
+    backup: true
+    regexp: "{{ item.regexp }}"
+    line: "{{ item.line }}"
+    state: "{{ item.state }}"
+  with_items:
+    - "{{ SSHD_CHANGES }}"
+    - { regexp: "{{ PORT_ENTRY_REGEXP }}", line: "{{ PORT_ENTRY_LINE }}", state: 'present' }
+  tags: sshd-config
+
+- name: Append AllowGroups Line to /etc/ssh/sshd_config
```

Comment:
@c-mart I kept this in a separate task because it does not use a `regexp`. But I suppose just setting `regexp: ""` might work since `lineinfile` defaults to EOF so I will give that a try.